Tools/NTupleReader.cc

```diff
@@ -267,11 +318,16 @@ bool NTupleReader::goToEventInternal(int evt, const bool filter)
     bool passFilters = false;
     do
     {
+        clearDerivedVectors();
```

Comment:
What vectors is this function deleting that where not already being deleted? Is this to clear up vectors created specifically from NanoAOD arrays?

Reply:
All these vectors are already being deleted, however, I moved where they were deleted to a fixed point in the nect event cycle so that I could check if they had ever been set for every event.  It's no extra work for NTupleReader, but it is more robust as now we can tell (internally to NTupleReader) when a vector has never been set in an event.